app/classes/query/modules/ordering.rb

```diff
@@ -148,7 +148,6 @@ def initialize_order_specs(by)
       "collection_numbers.name ASC, collection_numbers.number ASC"
 
     when "code"
-      where << "herbaria.code != ''"
       "herbaria.code ASC"
```

Comment:
Proposal:

"IF(COALESCE(herbaria.code, '') = '', 1, 0) ASC, herbaria.code ASC"

Looks like this works as desired.